homeassistant/components/demo/air_quality.py

```diff
@@ -32,39 +32,27 @@ async def async_setup_entry(
 class DemoAirQuality(AirQualityEntity):
     """Representation of Air Quality data."""
 
-    def __init__(self, name, pm_2_5, pm_10, n2o):
+    _attr_attribution = "Powered by Home Assistant"
+    _attr_should_poll = False
+
+    def __init__(self, name: str, pm_2_5: int, pm_10: int, n2o: int | None) -> None:
         """Initialize the Demo Air Quality."""
-        self._name = name
+        self._attr_name = f"Demo Air Quality {name}"
```

Comment:
I have also implemented class attributes on `attribution` and `should_poll`, as well as instance attributes on `name`.

If you feel that it should not be implemented on the `demo` component, then I can remove.
If you feel that it should be kept for a separate PR then I can split it out.